File: oop/heritage/Tp3/src/Etudiant.java
public class Etudiant extends Person {
    long num_card;

    public Etudiant(String lastname, String firstname, long num_card) {
        super(lastname, firstname);
        this.num_card = num_card;
    }

    public long getNum_card() {
        return num_card;
    }

    public void setNum_card(long num_card) {
        this.num_card = num_card;
    }

    @Override
    public String toString() {
        return "Etudiant [num_card=" + num_card + ", Id=" + getId() + ", lastname=" + getLastname() + ", firstname="
                + getFirstname() + "]";
    }
}

File: oop/heritage/CPP/exercice2/src/Etudiant.cpp
#include "Etudiant.hpp"

Etudiant::Etudiant(const std::string& lastname, const std::string& firstname, long long numCard)
    : Person(lastname, firstname), numCard(numCard) {}

long long Etudiant::getNumCard() const {
    return numCard;
}

void Etudiant::setNumCard(long long newNumCard) {
    numCard = newNumCard;
}

std::string Etudiant::toString() const {
    return Person::toString() + "Etudiant [num_card=" + std::to_string(numCard) + "]";
}
